### PR TITLE
fix sign up bug and user experience enhancement

### PR DESCRIPTION
**Pull request by spidemen**
1.  fix verify code send bug, need pass username check before send a email code 
or pass both username and email check before send phone verify code
2. add email length check
3.  merge from fix_register_progress

### CODE REVIEW

File: src/locales/zh.json [hunk 1]
```diff
@@ -56,7 +56,9 @@
         "You are doing that too much, please try again tomorrow",
     "error_confirmation_code_required": "'请输入你收到的验证码",
     "error_country_code_required": "请选择你的地区",
+    "error_username_invalid": "请首先输入一个邮箱有户名",
```
Comment (ety001) [inline src/locales/zh.json:59]: “请输入用户名”

File: src/components/Form/Signup/UserInfo.js [hunk 2]
```diff
@@ -123,8 +134,22 @@ class UserInfo extends React.Component {
...
-        if (this.state.email_code_sending) return;
         const { intl, locale } = this.props;
+        // check username first
+        if (!this.state.valid_username) {
+            this.props.form.setFields({
+                username: {
+                    value: '',
+                    errors: [
+                        new Error(
+                            intl.formatMessage({ id: 'error_username_invalid' })
```
Comment (ety001) [inline src/components/Form/Signup/UserInfo.js:145]: I think to use exist i18n item is enough.  Like `error_api_username_required`.

File: src/locales/zh.json (hunk 1 above)
Comment (spidemen) [reply src/locales/zh.json:59]: this code most is being use when a users click send email code without a valid username, so we need to tell the user that he or she need to input a valid username first not like 'input a username' though somehow it make sense

File: src/components/Form/Signup/UserInfo.js [hunk 3]
```diff
@@ -174,8 +199,37 @@ class UserInfo extends React.Component {
...
+                        ),
+                    ],
+                },
+            });
+            return;
+        }
+        // check email second
+        if (!this.state.valid_email) {
```
Comment (ety001) [inline src/components/Form/Signup/UserInfo.js:218]: why to check email in the SendPhoneCode progress? I think these are two separate events.

File: src/locales/zh.json (hunk 1 above)
Comment (ety001) [reply src/locales/zh.json:59]: This error message pops up when username input is empty. So just tell user that need to input a username is enough. Keep error message simple, not more adj. words. User is smart.
And also this error message has incorret word (邮箱有户名).

**Pull request closed without merging**